sql/core/src/main/java/org/apache/spark/sql/execution/BufferedRowIterator.java

```diff
@@ -73,14 +78,21 @@ public void append(InternalRow row) {
     currentRows.add(row);
   }
 
+  /**
+   * Sets the flag of stopping the query execution early.
+   */
+  public void setStopEarly(boolean value) {
```

Comment:
can we have more documents about how to use it? For now I see 2 use cases:
1. limit operator should call it with `true` when the limit is hit
2. blocking operator(sort, agg, etc.) should call it with `false` to reset it.

Reply:
Ok. Let me add it.

Reply:
You also hint me that we should reset stop early flag in sort exec node too. I will add it and related test.